hudi-client/hudi-client-common/src/main/java/org/apache/hudi/config/HoodieWriteConfig.java

```diff
@@ -2040,15 +2041,19 @@ public boolean useBloomIndexBucketizedChecking() {
   }
 
   public boolean isMetadataBloomFilterIndexEnabled() {
-    return isMetadataTableEnabled() && getMetadataConfig().isBloomFilterIndexEnabled();
+    return isMetadataTableEnabled() && getMetadataConfig().isBloomFilterIndexEnabled() || !isDropMetadataIndex(MetadataPartitionType.BLOOM_FILTERS.getPartitionPath());
```

Comment:
why OR condition. it should be AND right?
can you update PR desc w/ more illustration. 
how does one set mdt configs for inline and async flows. and how does deletion of indexes work prior to this patch. and how does it work w/ this patch

Reply:
also, can you add java docs for the condition we are checking here. 
i.e. 
either a given index is enabled explicitly in write config OR/AND if its not part of explicit delete list

Reply:
done both.